src/webgpu/shader/execution/expression/call/builtin/modf.spec.ts

```diff
@@ -18,11 +18,98 @@ Returns the result_struct for the given type.
 
 import { makeTestGroup } from '../../../../../../common/framework/test_group.js';
 import { GPUTest } from '../../../../../gpu_test.js';
-import { allInputSources } from '../../expression.js';
+import { f32, toVector, TypeF32, TypeVec } from '../../../../../util/conversion.js';
+import { modfInterval } from '../../../../../util/f32_interval.js';
+import { fullF32Range, quantizeToF32, vectorTestValues } from '../../../../../util/math.js';
+import { makeCaseCache } from '../../case_cache.js';
+import { allInputSources, Case, ExpressionBuilder, run } from '../../expression.js';
 
 export const g = makeTestGroup(GPUTest);
 
-g.test('scalar_f32')
+/* @returns an ExpressionBuilder that evaluates modf and returns .whole from the result structure */
+function wholeBuilder(): ExpressionBuilder {
```

Comment:
Slightly concerned that we've had to double the number of tests for the two structure members here, but I can see why you've done this.
When this is landed, perhaps I can take a look at adding structure support to the type / value framework.

Reply:
@zoddicus :  I suggest you file a followup issue for this enhancement.

Reply:
Done, https://github.com/gpuweb/cts/issues/2028